wild_lib/src/elf_writer.rs

```diff
@@ -1661,6 +1734,89 @@ fn apply_relocation(
     Ok(next_modifier)
 }
 
+fn apply_debug_relocation(
+    object_layout: &ObjectLayout,
+    offset_in_section: u64,
+    rel: &elf::Rela,
+    section_info: SectionInfo,
+    layout: &Layout,
+    out: &mut [u8],
+) -> Result<()> {
+    let section_address = section_info.section_address;
+    let place = section_address + offset_in_section;
+    let _span = tracing::trace_span!("debug relocation", address = place).entered();
+
+    let e = LittleEndian;
+    let symbol_index = rel
+        .symbol(e, false)
+        .context("Unsupported absolute relocation")?;
+    let sym = object_layout.object.symbol(symbol_index)?;
+    let section_index = object_layout.object.symbol_section(sym, symbol_index)?;
+
+    let addend = rel.r_addend.get(e) as u64;
+    let r_type = rel.r_type(e, false);
+    let rel_info = RelocationKindInfo::from_raw(r_type)?;
+
+    let resolution = layout
+        .merged_symbol_resolution(object_layout.symbol_id_range.input_to_id(symbol_index))
+        .or_else(|| {
+            section_index
+                .and_then(|section_index| object_layout.section_resolutions[section_index.0])
+        });
+
+    let value = if let Some(resolution) = resolution {
+        match rel_info.kind {
+            RelocationKind::Absolute => resolution.value_with_addend(
+                addend,
+                symbol_index,
+                object_layout,
+                &layout.merged_strings,
+                &layout.merged_string_start_addresses,
+            )?,
+            RelocationKind::DtpOff => resolution
+                .value()
+                .wrapping_sub(layout.tls_end_address())
+                .wrapping_add(addend),
+            _ => todo!(),
```

Comment:
It'll be quicker to figure out what's going on here if we instead do something like: `kind => bail!("Unsupported debug relocation kind {kind:?}")`